portal-backend/depmap/data_explorer_2/views.py

```diff
@@ -444,49 +441,56 @@ def unique_values_or_range():
     )
 
 
-@blueprint.route("/evaluate_context", methods=["POST"])
+@blueprint.route("/context/labels", methods=["POST"])
 @csrf_protect.exempt
-def evaluate_context():
+def get_labels_matching_context():
+    """
+    Get the full list of labels (in any dataset) which match the given context.
+    Also include the count of "candidate" labels, which belong to the context's 
+    dimension type (called "context_type" here).
+    """
     inputs = request.get_json()
     context = inputs["context"]
-    summarize = inputs["summarize"]
     context_type = context["context_type"]
-
     context_evaluator = ContextEvaluator(context)
     input_labels = get_entity_labels_across_datasets(context_type)
 
-    if summarize:
-        num_matches = sum(int(context_evaluator.is_match(x)) for x in input_labels)
-        return make_gzipped_json_response(
-            {"num_candidates": len(input_labels), "num_matches": num_matches}
-        )
-
-    labels = []
-
+    labels_matching_context = []
     for label in input_labels:
         if context_evaluator.is_match(label):
-            labels.append(label)
+            labels_matching_context.append(label)
+    aliases = get_aliases_matching_labels(context_type, labels_matching_context)
 
-    aliases = get_aliases_matching_labels(context_type, labels)
+    # Aliases should be removed from this response once they're no longer needed by context manager.
+    # The breadbox implementation of this endpoint will not return aliases.
+    response = {
+        "labels": labels_matching_context,
+        "aliases": aliases,
```

Comment:
Fyi, here is the one place that these "aliases" are still used in the frontend: https://github.com/broadinstitute/depmap-portal/blob/f7aadc1df076b840b059a3e0900dca15da1d1ed1/frontend/packages/portal-frontend/src/data-explorer-2/components/ContextManager/DownloadContextModal.tsx#L59

@rcreasi I think I remember you saying that this is something that could be replaced with a call to a different endpoint - is that right? If so, I can make a follow-up task in Asana.

Reply:
Sort of, with the current implementation:
* For features, the actual labels are returned
* For samples, the given_ids are returned as `labels` (this is the part we were talking about earlier), and I believe that's why the "aliases" are used. 

If I remember correctly, the reason we do this is because "sample labels" historically, were not easily supported in the legacy backend. Luckily, I think we're at a point now where:
1. The portal-backend could return actual sample labels without too much effort
2. It will be easy for breadbox to return actual sample labels

Maybe I'll add a topic to our Breadbox meeting to talk about this more. At some point, soon we should make the switch to returning actual sample labels - and perhaps that needs to happen before these aliases can be replaced.

Reply:
Yes, there's a clear path to making the frontend no longer rely on aliases. There's already this `/get_metadata` endpoint.

https://github.com/broadinstitute/depmap-portal/blob/f7aadc1df076b840b059a3e0900dca15da1d1ed1/portal-backend/depmap/data_explorer_2/views.py#L153

You give it a slice ID and get back indexed values. That's what powers the "color by property" functionality. We could just leverage that to get a mapping from DepMap ID to cell line name by using it with the slice ID `slice/cell_line_display_name/all/label`.